app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # Guest user
+
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :read, :all
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, author_id: user.id
```

Comment:
- [ ] Hi @ezraermy, as you can see on the record below, a registered user can't create a post. When I click on New Post button, the app didn't works well. A user should be able to create a post and comment so that we should be able to test other functionalities like assigning an "admin" role to a user and testing if he can manage(delete) the posts and comments of other users.


[Screencast from 08-31-2023 07:34:53 AM.webm](https://github.com/ezraermy/rails-blog/assets/63087491/f11b28d2-9771-4265-8f68-cba39f82c5e4)

- [ ] At this point we are supposed to implement destroy actions in the PostsController and in the CommentsController. Kindly do so so that other functionalities for this Milestone should be tested. A user should be able to delete his own posts and comments. If the user is an assigned an admin role, he should be able to delete the posts and comments of other users. This is why the delete action should be implemented for both comments and posts controller :+1: